### ISSUE TITLE
Scheduled maintenance

### ISSUE THREAD

**Issue by jbrooksuk** [cachethq/Cachet#112]
What do we think about adding a separate status level for "scheduled". I'm not 100% sure on this since we could also add a date box on the incident page which can be set in the future. The status page will then detect future dates and list them appropriately as scheduled.

**Comment by jbrooksuk** [comment]
Does anybody have an idea on how this should work? What's the best way to show a scheduled status?

**Comment by joecohens** [comment]
I think we can put a publish_date with a datepicker and kind of stop using the created_at value. In this way we can change the order of incidents too.

**Comment by jbrooksuk** [comment]
@joecohens good idea. Incidents and scheduled maintenance could be merged into one by just adding a datepicker. When the `publish_at` date is in the future, we can show a "non-timeline" view of incidents (like we used to have) marked as **Scheduled Maintenance**.

The thing is, we'd want to continue showing the schedules until it's complete, so maybe we do need it as a separate thing?

**Comment by jbrooksuk** [comment]
The table could remain the same just with some flags. But separating them would mean we don't clutter the existing incidents dashboard.

**Comment by jeanfrancis** [comment]
why not also automatically notify users in advance when it starts and once the service is restored

**Comment by jbrooksuk** [comment]
This is something we'll be able to integrate once #316 is done.

**Comment by jbrooksuk** [comment]
This is being worked on in #404 - but it'll be known as "Scheduled maintenance" which is still an incident, but with a `publish_date`.
